be/src/util/jni-util.cpp

```diff
@@ -0,0 +1,244 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "util/jni-util.h"
+
+#include <jni.h>
+#include <stdlib.h>
+
+#include "gutil/once.h"
+#include "gutil/strings/substitute.h"
+
+using std::string;
+
+namespace doris {
+
+namespace {
+JavaVM* g_vm;
+GoogleOnceType g_vm_once = GOOGLE_ONCE_INIT;
+
+void FindOrCreateJavaVM() {
+    int num_vms;
+    int rv = JNI_GetCreatedJavaVMs(&g_vm, 1, &num_vms);
+    if (rv == 0) {
+        LOG(INFO) << "Create first JVM";
+        JNIEnv *env;
+        JavaVMInitArgs vm_args;
+        JavaVMOption options[1];
+        std::stringstream ss;
+        char* str = getenv("DORIS_JNI_CLASSPATH_PARAMETER");
+        options[0].optionString = str;
+        vm_args.version = JNI_VERSION_1_8;
+        vm_args.options = options;
+        vm_args.nOptions = 1;
+        vm_args.ignoreUnrecognized = JNI_TRUE;
+
+        int res = JNI_CreateJavaVM(&g_vm, (void **)&env, &vm_args);
+        DCHECK_LT(res, 0) << "Failed tp create JVM, code= " << res;
+    } else {
+        CHECK_EQ(rv, 0) << "Could not find any created Java VM. Must init libhdfs first";
```

Comment:
libhdfs?

Reply:
Fixed it. libhdfs will create a JVM and we can reuse it after that but we use libhdfs++ instead.